src/parser/shared/modules/items/bfa/raids/battleofdazaralor/WardOfEnvelopment.js

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS/index';
+import ITEMS from 'common/ITEMS/index';
+import ItemLink from 'common/ItemLink';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import ItemHealingDone from 'interface/others/ItemHealingDone';
+import Abilities from 'parser/core/modules/Abilities';
+
+const MAX_ALLIES_HIT = 5;
+const ACTIVATION_COOLDOWN = 120; // seconds
+
+/**
+ * Use: Envelop up to 5 allies in the targeted area in shared shields for 10 sec, preventing up to [x] damage taken by any of them. Damage prevented is increased when affecting multiple allies. (2 Min Cooldown)
+ * 
+ * Allies affected have the Enveloping Protection buff, lasts 10 seconds or until all damage is absorbed.
+ * Hitting 1 ally gives the base value, each additional ally hit adds 7.5% of the base value. Maximum of 5 people hit for 30% extra on top of the base value.
+ */
+class WardOfEnvelopment extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+  }
+
+  absorbUsed = 0;
+  absorbWasted = 0;
+  targetsHit = 0;
+  firstExpiration = false;
+  uses = 0;
+  
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrinket(ITEMS.WARD_OF_ENVELOPMENT.id);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell([SPELLS.ENVELOPING_PROTECTION]), this.onCast);
+    this.addEventListener(Events.absorbed.by(SELECTED_PLAYER).spell([SPELLS.ENVELOPING_PROTECTION]), this.onAbsorb);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.ENVELOPING_PROTECTION]), this.onBuff);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.ENVELOPING_PROTECTION]), this.onRemoveBuff);
+
+    if (this.active) {
+      this.abilities.add({
+        spell: SPELLS.ENVELOPING_PROTECTION,
+        name: ITEMS.WARD_OF_ENVELOPMENT.name,
+        category: Abilities.SPELL_CATEGORIES.ITEMS,
+        cooldown: ACTIVATION_COOLDOWN,
+        castEfficiency: {
+          suggestion: true,
+        },
+      });
+    }
+  }
+
+  onCast(event) {
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.ENVELOPING_PROTECTION.id){
```

Comment:
Spell ID checks in all of these handlers are unnecessary and will always return true (the `spell()` method in `addEventListener()` takes care of filtering the spells)